pkg/gardenlet/controller/vpaevictionrequirements/reconciler.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vpaevictionrequirements
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	vpaautoscalingv1 "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	"k8s.io/utils/clock"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/gardener/gardener/pkg/gardenlet/apis/config"
+	"github.com/gardener/gardener/pkg/utils/timewindow"
+)
+
+var cpuAndMemory = []corev1.ResourceName{corev1.ResourceMemory, corev1.ResourceCPU}
+var upscaleOnlyRequirement = []*vpaautoscalingv1.EvictionRequirement{
+	{
+		Resources:         cpuAndMemory,
+		ChangeRequirement: vpaautoscalingv1.TargetHigherThanRequests,
+	},
+}
```

Comment:
Nit: `cpuAndMemory` seems to not be used somewhere else
```suggestion
var upscaleOnlyRequirement = []*vpaautoscalingv1.EvictionRequirement{{
	Resources:         []corev1.ResourceName{corev1.ResourceMemory, corev1.ResourceCPU},
	ChangeRequirement: vpaautoscalingv1.TargetHigherThanRequests,
}}
```